code/controllers/subsystem/weeds.dm

```diff
@@ -78,8 +78,6 @@ SUBSYSTEM_DEF(weeds)
 		return
 	var/obj/alien/weeds/weed_to_spawn = node.weed_type
 	var/swapped = FALSE
-	if(iswallturf(T))
-		weed_to_spawn = /obj/alien/weeds/weedwall
```

Comment:
Better to not remove this, since the functionality exists you're just turning it off for walls.

Reply:
Pointless check, why do we care if it's a wall if walls are unweedable?